src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PreferIsExactSpellingIsTrueForKnownApisAnalyzer.cs

```diff
@@ -0,0 +1,164 @@
+                                                                             isDataflowRule: false,
+                                                                             isEnabledByDefaultInFxCopAnalyzers: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DefaultRule, WideRule);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            analysisContext.RegisterCompilationStartAction(ctx =>
+            {
+                if (ctx.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeInteropServicesDllImportAttribute, out var dllImportType))
+                {
+                    ctx.RegisterSymbolAction(x => AnalyzeExternMethod(x, dllImportType), SymbolKind.Method);
+                }
+            });
+        }
+
+        private static void AnalyzeExternMethod(SymbolAnalysisContext context, INamedTypeSymbol dllImportType)
+        {
+            var methodSymbol = (IMethodSymbol)context.Symbol;
+            if (!methodSymbol.IsExtern)
+            {
+                return;
+            }
+
+            var dllImportAttribute = methodSymbol
+                .GetAttributes()
+                .FirstOrDefault(x => x.AttributeClass.Equals(dllImportType));
+
+            if (dllImportAttribute is null)
+            {
+                return;
+            }
+
+            var hasExactSpellingParameter = dllImportAttribute.NamedArguments.FirstOrDefault(x => x.Key.Equals("ExactSpelling", StringComparison.Ordinal));
+            var hasCharSetParameter = dllImportAttribute.NamedArguments.FirstOrDefault(x => x.Key.Equals("CharSet", StringComparison.Ordinal));
+
+            var methodName = methodSymbol.GetActualExternName(dllImportAttribute);
+            var isExactSpelling = hasExactSpellingParameter.Key is not null
+                && hasExactSpellingParameter.Value.Kind != TypedConstantKind.Array
+                && bool.TryParse(hasExactSpellingParameter.Value.Value?.ToString(), out var isExact)
+                && isExact;
+            var isCharSetUnicode = hasCharSetParameter.Key is not null
+                && hasCharSetParameter.Value.Kind != TypedConstantKind.Array
+                && Enum.TryParse<CharSet>(hasCharSetParameter.Value.Value?.ToString(), out var actualCharSet)
+                && actualCharSet == CharSet.Unicode;
+            var dllName = dllImportAttribute.ConstructorArguments.FirstOrDefault().Value?.ToString() ?? string.Empty;
+
+            if (!KnownApis.Value.TryGetValue(dllName, out var methods))
+            {
+                return;
+            }
+
+            if (methods.Contains(methodName))
+            {
+                if (isCharSetUnicode && !isExactSpelling && methodName.EndsWith("W", StringComparison.Ordinal))
```

Comment:
`!isExactSpelling` better to be moved before DB search ( before `if (methods.Contains(methodName))`), here and all other occurrences below 
```suggestion
                if (isCharSetUnicode && methodName.EndsWith("W", StringComparison.Ordinal))
```